packages/block-library/src/navigation-menu/edit.js

```diff
@@ -25,8 +27,18 @@ function NavigationMenu( {
 	attributes,
 	setAttributes,
 	clientId,
+	pages,
+	isRequesting,
 } ) {
 	const { navigatorToolbarButton, navigatorModal } = useBlockNavigator( clientId );
+	let defaultMenuItems = false;
```

Comment:
We are computing a new defaultMenuItems on each re-render because the map returns a different reference that in turn makes us pass a different template on each render. Changing a template involves complex logic and store update so we should avoid that.
We can solve the problem by wrapping this inside useMemo so the value is memoized.